wicket-core/src/main/java/org/apache/wicket/markup/head/AbstractJavaScriptReferenceHeaderItem.java

```diff
@@ -135,6 +135,7 @@ public boolean equals(Object o)
 	@Override
 	public int hashCode()
 	{
-		return Objects.hash(async, defer, charset);
+		return (charset != null ? charset.hashCode() << 2 : 0)
```

Comment:
Please use an auto-generated hash-code for this method as well. Doing everything consistently, makes it easier to simply regenerate the hash-code should a new field be added.